src/kagglehub/http_resolver.py

```diff
@@ -11,14 +11,64 @@
     mark_as_complete,
 )
 from kagglehub.clients import KaggleApiV1Client
-from kagglehub.handle import ModelHandle
+from kagglehub.handle import DatasetHandle, ModelHandle
 from kagglehub.resolver import Resolver
 
 MODEL_INSTANCE_VERSION_FIELD = "versionNumber"
 
 logger = logging.getLogger(__name__)
 
 
+class DatasetHttpResolver(Resolver[DatasetHandle]):
+    def is_supported(self, *_, **__) -> bool:  # noqa: ANN002, ANN003
+        # Downloading files over HTTP is supported in all environments for all handles / paths.
+        return True
+
+    def __call__(self, h: DatasetHandle, path: Optional[str] = None, *, force_download: Optional[bool] = False) -> str:
+        api_client = KaggleApiV1Client()
+
+        dataset_path = load_from_cache(h, path)
+        if dataset_path and not force_download:
+            return dataset_path  # Already cached
+        elif dataset_path and force_download:
+            delete_from_cache(h, path)
+
+        url_path = _build_dataset_download_url_path(h)
+        out_path = get_cached_path(h, path)
+
+        # Create the intermediary directories
+        if path:
+            # Downloading a single file.
+            os.makedirs(os.path.dirname(out_path), exist_ok=True)
+            api_client.download_file(url_path + "/" + path, out_path, h)
+        else:
+            # Downloading the full archived bundle.
```

Comment:
@jeward414 @jplotts (cc: @rosbo )

FYI: Vincent called out an edgecase issue with unversioned handles and a new dataset version that is processed, but without a zip generated yet.

https://docs.google.com/document/d/1eNmztP3xcTit_1m_LkWi5wFt7Z4mL-00lPAs-IQTtxw/edit?usp=drivesdk&resourcekey=0-sr7en0pAcVJaSljXfhNjQg&disco=AAABJp2zl44

tldr is that we will throw an exception if a version is still zipping. for some datasets with many files or large files, this can take up to 24 hours to zip.

I offered some ideas to address in that thread, but curious your thoughts.

Reply:
I think the idea of throwing an exception is good, instead of letting the user not getting the latest version when they didn't specify.